great_expectations/datasource/data_connector/runtime_data_connector.py

```diff
@@ -151,6 +149,10 @@ def _get_batch_definition_list_from_batch_request(
         self,
         batch_request: BatchRequestBase,
     ) -> List[BatchDefinition]:
+
```

Comment:
During BatchPod, we had an iteration, where setting `data_asset_name` on `RuntimeDataConnector` was implemented this way.  However, in addition, we had the `self._`data_asset_name` to store it.  This was disallowed -- the `RuntimeDataConnector` was to have no state.  If this is now allowed, we need to make sure that this property is persistent long enough to be available for wherever it is needed: validations, data_docs, Checkpoints, etc.  Are we sure that this implementation now assures the proper available of `data_asset_name` in `RuntimeDataConnector` as it is in the other connectors?  Which tests prove this?  Thanks!

Reply:
Yes, I remember this. This I can double-check this